sdk/search/Azure.Search/src/Serialization/JsonExtensions.cs

```diff
@@ -0,0 +1,473 @@
+            return name;
+        }
+
+        /// <summary>
+        /// Verify and read the next string value.
+        /// </summary>
+        /// <param name="reader">The JSON reader.</param>
+        /// <returns>The next string value.</returns>
+        public static string ExpectsString(this ref Utf8JsonReader reader)
+        {
+            reader.Expects(JsonTokenType.String);
+            return reader.GetString();
+        }
+
+        /// <summary>
+        /// Verify and read the next double? value.
+        /// </summary>
+        /// <param name="reader">The JSON reader.</param>
+        /// <returns>The next double? value.</returns>
+        public static double? ExpectsNullableDouble(
+            this ref Utf8JsonReader reader)
+        {
+            if (reader.TokenType == JsonTokenType.Null)
+            {
+                return null;
+            }
+            reader.Expects(JsonTokenType.Number);
+            return reader.GetDouble();
+        }
+
+        /// <summary>
+        /// Verify and read the next long? value.
+        /// </summary>
+        /// <param name="reader">The JSON reader.</param>
+        /// <returns>The next long? value.</returns>
+        public static long? ExpectsNullableLong(
+            this ref Utf8JsonReader reader)
+        {
+            if (reader.TokenType == JsonTokenType.Null)
+            {
+                return null;
+            }
+            reader.Expects(JsonTokenType.Number);
+            return reader.GetInt64();
+        }
+
+        /// <summary>
+        /// Deserialize a JSON stream.
+        /// </summary>
+        /// <typeparam name="T">
+        /// The target type to deserialize the JSON stream into.
+        /// </typeparam>
+        /// <param name="json">A JSON stream.</param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate notifications
+        /// that the operation should be canceled.
+        /// </param>
+        /// <returns>A deserialized object.</returns>
+        public static async Task<T> DeserializeAsync<T>(
+            this Stream json,
+            CancellationToken cancellationToken)
+        {
+            if (json == null)
+            {
+                return default;
+            }
+            else
+            {
+                return await JsonSerializer.DeserializeAsync<T>(
+                    json,
+                    JsonExtensions.SerializerOptions,
+                    cancellationToken)
+                    .ConfigureAwait(false);
+            }
+        }
+
+        /// <summary>
+        /// Deserialize a JSON stream.
+        /// </summary>
+        /// <typeparam name="T">
+        /// The target type to deserialize the JSON stream into.
+        /// </typeparam>
+        /// <param name="json">A JSON stream.</param>
+        /// <returns>A deserialized object.</returns>
+        public static T Deserialize<T>(this Stream json)
```

Comment:
Why don't you `CopyTo` a memory stream here?

Reply:
I borrowed this code at Ahson's suggestion when we were talking about doing a CopyTo MemoryStream.

Reply:
I really wonder if this amount of complexity is warranted here and if it's worth the support cost later...

Reply:
This is a fair question.  I'm going to leave it today and have that conversation with you and Ahson in the near future when I pull this into Shared Source since anyone else supporting custom user schemas is going to need it.